tt_metal/llrt/rtoptions.hpp

```diff
@@ -56,6 +58,14 @@ class RunTimeOptions {
     inline void set_watcher_dump_all(bool dump_all)   { watcher_dump_all = dump_all; }
     inline int get_watcher_append()                   { return watcher_append; }
     inline void set_watcher_append(bool append)       { watcher_append = append; }
+    inline bool get_watcher_catch_exceptions()        { return watcher_catch_exceptions; }
+    inline void set_watcher_catch_exceptions(bool catch_exceptions) {
```

Comment:
thinking we should collapse the 2 into 1: ie, we're either in "test mode" in which case we catch exceptions or we're in "run mode" in which case we throw exceptions.  having this per-concept is likely overkill (and possibly more maintenance)

Reply:
Good point, changed this to a single setting in latest rebase